style: prefer `import type`

Alfred-Skyblue opened the pull request:
When we intend to use a value solely as a type, we should import the type using `import type` instead of importing the value. I have only handled cases where values are used as types, perhaps we should set `"importsNotUsedAsValues": "error"` to prevent such occurrences?

sxzz reviewed (commented):
I think the best approach to maintaining consistency is by implementing an ESLint rule.

Alfred-Skyblue commented:
> I think the best approach to maintaining consistency is by implementing an ESLint rule.

I added `@typescript-eslinteslint-plugin` dependency and added ESLint rule

liuseen-l commented:
LGTM

sxzz commented:
Conflicted. Could you please rebase your branch?

Alfred-Skyblue commented:
> Conflicted. Could you please rebase your branch?

Hi, I have resolved the conflict on the branch

Alfred-Skyblue commented:
This PR will cause conflicts with most of the PRs, it is recommended to be merged at the end.

sxzz commented:
It will create lots of conflicts for other PRs. We need to wait for a time to merge/clean up the PR. Let's put that on hold for now. :innocent:

Alfred-Skyblue commented:
Yes, this PR will cause a large conflict, we should put it to the last merge.

Alfred-Skyblue commented:
In PR #8978, the `@typescript-eslint/eslint-plugin` was added, which can also introduce rules for using `import type`, and it offers additional features. Therefore, the functionality of this PR will be merged into #8978. We will temporarily close this PR.

cc @sxzz
